canDetectLibC should check target Target.abi

### Zig Version

0.14.0-dev.1827+e1e151df0

### Steps to Reproduce and Observed Behavior

```
$zig libc
```
Error:
![bug1](https://github.com/user-attachments/assets/0a99229c-bdff-4221-a38c-e46548d07589)

```
/Users/dk/src/zig/zig/lib/std/Target/Query.zig:377:38: error: no field or member function named 'isAndroid' in '?Target.Abi'
        if (os == .linux and self.abi.isAndroid()) return true;
                             ~~~~~~~~^~~~~~~~~~
referenced by:
    main: /Users/dk/src/zig/zig/lib/compiler/libc.zig:116:40
    main: /Users/dk/src/zig/zig/lib/std/start.zig:621:37
    3 reference(s) hidden; use '-freference-trace=5' to see all references
```

### Expected Behavior

should not be error

--- a/lib/std/Target/Query.zig
+++ b/lib/std/Target/Query.zig
@@ -375,7 +375,7 @@ pub fn canDetectLibC(self: Query) bool {
     if (self.os_tag) |os| {
         if (builtin.os.tag == .macos and os.isDarwin()) return true;
         if (os == .linux) {
-            if (self.abi) |abi| return abi.isAndroid();
+            if (self.abi) |abi| if (abi.isAndroid()) return true;
         }
     }
     return false;

--- a/lib/std/Target/Query.zig
+++ b/lib/std/Target/Query.zig
@@ -374,8 +374,8 @@ pub fn canDetectLibC(self: Query) bool {
     if (self.isNativeOs()) return true;
     if (self.os_tag) |os| {
         if (builtin.os.tag == .macos and os.isDarwin()) return true;
-        if (os == .linux ) {
-            if (self.abi) |abi|  return abi.isAndroid();
+        if (os == .linux) {
+            if (self.abi) |abi| return abi.isAndroid();
         }
     }
     return false;

--- a/lib/std/Target/Query.zig
+++ b/lib/std/Target/Query.zig
@@ -374,7 +374,7 @@ pub fn canDetectLibC(self: Query) bool {
     if (self.isNativeOs()) return true;
     if (self.os_tag) |os| {
         if (builtin.os.tag == .macos and os.isDarwin()) return true;
-        if (os == .linux and self.abi.?.isAndroid()) return true;
+        if (os == .linux and self.abi != null and self.abi.?.isAndroid()) return true;
     }
     return false;
 }

--- a/lib/std/Target/Query.zig
+++ b/lib/std/Target/Query.zig
@@ -374,7 +374,7 @@ pub fn canDetectLibC(self: Query) bool {
     if (self.isNativeOs()) return true;
     if (self.os_tag) |os| {
         if (builtin.os.tag == .macos and os.isDarwin()) return true;
-        if (os == .linux and self.abi.isAndroid()) return true;
+        if (os == .linux and self.abi.?.isAndroid()) return true;
     }
     return false;
 }

--- a/lib/std/Target/Query.zig
+++ b/lib/std/Target/Query.zig
@@ -374,7 +374,9 @@ pub fn canDetectLibC(self: Query) bool {
     if (self.isNativeOs()) return true;
     if (self.os_tag) |os| {
         if (builtin.os.tag == .macos and os.isDarwin()) return true;
-        if (os == .linux and self.abi != null and self.abi.?.isAndroid()) return true;
+        if (os == .linux ) {
+            if (self.abi) |abi|  return abi.isAndroid();
+        }
     }
     return false;
 }

